main.go

```diff
@@ -268,10 +272,22 @@ func generate(in configInput) ([]byte, error) {
 		// Authorities with an empty server config will end up using
 		// the top-level server config. For more details, see:
 		// https://github.com/grpc/proposal/blob/master/A47-xds-federation.md#bootstrap-config-changes.
-		c.Authorities = map[string]struct{}{
+		c.Authorities = map[string]Authority{
 			tdURI: {},
 			"":    {},
 		}
+		if in.includeC2PAuthority {
+			c.Authorities[c2pAuthority] = Authority{
+				XdsServers: []server{
+					{
+						ServerUri: "dns:///directpath-pa.googleapis.com",
+						ChannelCreds: []creds{
+							{Type: "google_default"},
```

Comment:
We need server features with `v3` support for c2p TD as well.

Reply:
sure, updated. So do we also want to check for `ignore_resource_deletion` too?

Reply:
Hmm ... maybe it might be a good idea to pull this out into a helper function. That way, we can be sure that we use the same logic to generate the server config for both TDs.